pyteal/ast/itxn.py

```diff
@@ -225,14 +225,17 @@ def MethodCall(
             app_id: An expression that evaluates to a `TealType.uint64` corresponding to the application being called.
             method_signature: A string representing the method signature of the method we're calling. This is used to do
                 type checking on the arguments passed and to create the method selector passed as the first argument.
-            args: A list of arguments to pass to the application. These args may be one of three types
-                1. An ABI type: Any ABI type other than ReferenceTypes (asset,application,account) and TransactionTypes. These are encoded
-                    as part of transaction preparation. The type passed _MUST_ match the type specified in the `method_signature` passed.
-                2. An Expr: An expression that evaluates to bytes representing a valid ABI encoded type passed directly to the arguments array.
-                    No type checking is performed besides checking that it evaluates to `TealType.bytes`.
-                3. A dictionary: A dictionary containing TxnField to Expr that describe Transactions to be pre-pended to the transaction group being constructed.
-                    The `TxnField.type_enum` key MUST be set and MUST match the expected transaction type specified in the `method_signature`.
-            fields: A dictionary whose keys are fields to set and whose values are the value each
+            args: A list of arguments to pass to the application. The values in this list depend on the kind of argument you wish to pass:
+
+                - For basic ABI arguments (not Reference or Transaction types): The ABI type is passed directly and encoded as part of transaction preparation. The type passed _MUST_ match the type specified in the `method_signature` passed.
+
+                - For Reference arguments: Either the Reference type or an Expr that returns the type corresponding to the reference type are allowed (Asset:TealType.uint64, Application:TealType.uint64, Account:TealType.bytes).
+
+                - For Transaction arguments: A dictionary containing TxnField to Expr that describe Transactions to be pre-pended to the transaction group being constructed.  The `TxnField.type_enum` key MUST be set and MUST match the expected transaction type specified in the `method_signature`.
+
+                - For any others: An expression that evaluates to bytes representing a valid ABI encoded type passed directly to the arguments array.  No type checking is performed besides checking that it evaluates to `TealType.bytes`.
```

Comment:
I think this would be more appropriate under basic ABI arguments, since it doesn't apply to reference or transaction types?